core/src/main/scala/filodb.core/binaryrecord2/RecordBuilder.scala

```diff
@@ -132,6 +134,58 @@ final class RecordBuilder(memFactory: MemFactory,
     fieldNo += 1
   }
 
+  /**
+    * IMPORTANT: Internal method, does not update hash values for the map key/values individually.
+    * If this method is used, then caller needs to also update the partitionHash manually.
+    */
+  private def addMap(base: Any, offset: Long, numBytes: Int): Unit = {
+    require(numBytes < 65536, s"bytes too large ($numBytes bytes) for addBlob")
+    checkFieldAndMemory(numBytes + 4)
+    UnsafeUtils.setInt(curBase, curRecEndOffset, numBytes) // length of blob
+    UnsafeUtils.unsafe.copyMemory(base, offset, curBase, curRecEndOffset + 4, numBytes)
+    updateFieldPointerAndLens(numBytes + 4)
+    fieldNo += 1
+  }
+
+  private def addStringFromBr(base: Any, offset: Long, col: Int, schema: RecordSchema): Unit = {
+    val strDataOffset = schema.utf8StringOffset(base, offset, col)
+    addBlob(base, strDataOffset + 2, schema.blobNumBytes(base, strDataOffset, col))
+  }
+
+  /**
+    * IMPORTANT: Internal method, does not update hash values for the map key/values individually.
+    * If this method is used, then caller needs to also update the partitionHash manually.
+    */
+  private def addMapFromBr(base: Any, offset: Long, col: Int, schema: RecordSchema): Unit = {
+    val strDataOffset = schema.utf8StringOffset(base, offset, col)
+    addMap(base, strDataOffset + 4, BinaryRegionMedium.numBytes(base, strDataOffset))
+  }
+
+  private def addLongFromBr(base: Any, offset: Long, col: Int, schema: RecordSchema): Unit = {
+    addLong(schema.getLong(base, offset, col))
+  }
+
+  private def addDoubleFromBr(base: Any, offset: Long, col: Int, schema: RecordSchema): Unit = {
+    addDouble(schema.getDouble(base, offset, col))
+  }
+
+  /**
+    * This also updates the hash for this record. OK since partKeys are added at the very end
```

Comment:
Let's add a more extensive ScalaDoc here.  Also I think a better name for the method is `addPartKeyRecordFields`; since we are adding all the fields from a partition key BR.